src/main/java/com/contentgrid/gateway/collections/ConcurrentLookup.java

```diff
@@ -116,5 +139,47 @@ public void remove(T data) {
             var list = this.data.get(key);
             list.remove(data);
         }
+
+        @Override
+        public void clear() {
+            this.data.clear();
+        }
+    }
+
+    private static class MultiIndex<L, T> implements Index<L, T> {
+
+        private final Map<L, List<T>> data = new HashMap<>();
```

Comment:
This implementation is not thread-safe (and neither is the one for `LookupIndex`